### ISSUE TITLE
Wrong formatting in special case

### ISSUE THREAD

**Issue by xdev1** [prettier/prettier#13237]
**Prettier 2.7.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVAAgAoBs4CGAznJgBYEBupBmuEEA1pgTJjGaQCYCWAZnzgAnBDAA6UTFIAOEIjxg9omCH0xiQADwCeALw2YekgEYQOmakPnQimdKgkSMF4dckBGOw6gS4m2UJsXHB8BACuuGwA7hTwlgAUfGFQYIrK8QRImMASUgT5BQVZRDBCRgDmANxSuZjG9Q0NxaUVlbVgHZ2dzWVQVRIAvgCU2bUiMGFCklARuG1QAwA06lp6GkPzTliWbpgATJjx0LjamLwCwghgpDy2HKyYAAaoqI8j9o5QfgFBIeGRmBirDgCSSKTSkgyWRykhYhUKPVaUhetUaaMRfXmUi6OIx-QWIxhUnGk2ms3mSxWOn0IA2EhAixAEGkEKIyFABCEQggUWwnIQbJQBFwUQI2jZjOMQgIYEYcBgAGUCABbOAAGSMcGQoVwJEl0tl8oV0hlFWQpTCcEZcGVxjgXGCXDVBD6YQI5TgADEIEJlaxFH1kCBwjAIAyQGQYMrcAB1MgKOBEE3XBUChQ8SgKbRBsBECUgIwkQLYaXlP3a4V6kAAKyImgVFXwAEUwmYtUgdVWTVZhEGYNppImwGUWeHpL0YDGeFwOMgABwABkZ47kcBj0ukQfHieE1HDAEdW-AS8zBcGiABaL72+3hkSHngiEvu8sdytWkAkZU8c1CS2M+Q+mbI9207D8YAIYwpxnMhkD2RlSgIHhcAqABhCBlVfEBEwAVnDMISAAFUgwUwMZShLQASSgYJYAVYceBZABBGiFX7fAK11OABgGIA)

```sh
--parser typescript
```

**Input:**

```tsx
/* Please have a look at the different
   position of "xyz" in both versions */

/* version 1 */

export default whatever(function (a: {
  aaaaaaaaa: string;  
  bbbbbbbbb: string;
  ccccccccc: string;
}) {
  return null;
}, "xyz");

/* version 2 (only difference is that `//`) */

export default whatever(function (a: {
  aaaaaaaaa: string;  //
  bbbbbbbbb: string;
  ccccccccc: string;
}) {
  return null;
}, "xyz");

```

**Output:**

```tsx
/* Please have a look at the different
   position of 'xyz' in both versions */

/* version 1 */

export default whatever(function (a: {
  aaaaaaaaa: string;
  bbbbbbbbb: string;
  ccccccccc: string;
}) {
  return null;
},
"xyz");

/* version 2 (only difference is that `//`) */

export default whatever(function (a: {
  aaaaaaaaa: string; //
  bbbbbbbbb: string;
  ccccccccc: string;
}) {
  return null;
}, "xyz");

```

**Expected behavior:**

Version 1 should look like version 2, shouldn't it?

**Comment by thorn0** [comment]
Not TypeScript-specific:

**Prettier 2.7.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVAAgG5wE4DOAltJgIybqoA6UtcAHgA4R4wAicAZgIYCuAGxgB1ABY94uPAAoufKGBgkomabUyYeW7Tp4BlGHiJQA5gBp1mAEY3bdqwaOmLKzGHcfPYR8fO0AlJjAlnhwMHx4KlCCAgDctAC+ZpjUIAwAngBeqf7xdFAYOPjEpABMqtAC6ZgAJkRcXPgIYHCYRASYMOIwmAAGqKi9gVS09MysHNz8QmIScFKy8orKqpa66z7OGgOW9nubfq5exwcugcGuoeGRmNECcYnJqRnZILm0IGYgEExK0ATIUA8PB4CAAdwACsCEACUDwBGCeOkAV8rHgeGAANZhPQ8AC2cAAMsY4MheAICHBUeisTimBjfMhDHwqSA4HirHAajUuYSeKY+DwTHAAGKsPESJSmZAgfgwCCfECiGB4gRiIjwAj0lp6GEaojYDXpGVgAgokDGSlsCHokwSsnwylfABWBAYel8AjgAEU+BB4A6Kaz6YR8DKrDxOQJFUwnCIiDUusgABwABi+sYglOE6KYMtjcCtuEVAEc-fAbT9YbKCABaKBwLlcxWhMtEUI2oX2pDkp0gSl4ohMvAsr7EUxe33+0k9x2smCR4QJpNIUpfQw8IgCXwAYQgeO7bIIAFZFXxKQAVSOw3us7AsgCSUB5sD0YCMvwAgs+DOkvYHKQSBIgA)

```sh
--parser babel
```

**Input:**

```jsx
/* version 1 */

exportDefaultWhatever(function (
  aaaaaaaaaaaString,
  bbbbbbbbbbbString,
  cccccccccccString,
) {
  return null;
}, "xyz");

/* version 2 (only difference is that `//`) */

exportDefaultWhatever(function (
  aaaaaaaaaaaString,  //
  bbbbbbbbbbbString,
  cccccccccccString,
) {
  return null;
}, "xyz");

```

**Output:**

```jsx
/* version 1 */

exportDefaultWhatever(function (
  aaaaaaaaaaaString,
  bbbbbbbbbbbString,
  cccccccccccString
) {
  return null;
},
"xyz");

/* version 2 (only difference is that `//`) */

exportDefaultWhatever(function (
  aaaaaaaaaaaString, //
  bbbbbbbbbbbString,
  cccccccccccString
) {
  return null;
}, "xyz");

```

**Comment by thorn0** [comment]
fixed by #13341
